lib/draw_badge_api.ts

```diff
@@ -39,66 +41,85 @@ export class DrawBadgeApi {
     this.handwave = handwave;
     return this.forceDrawing();
   }
+
+  drawCameraSizeChange(videoWidth: number, videoHeight: number) {
+    this.videoWidth = videoWidth;
+    this.videoHeight = videoHeight;
+    // console.log('it finally changed!!!!!!')
+    // console.log(videoWidth, videoHeight)
+    return this.forceDrawing();
+  }
 }
 
 // TODO: make sure the imageData scale and resize correctly based on window size.
 //       make need to make some more Zoom API calls to get user window size.
-function drawEverythingToImage(nametag: NameTagBadge, handWave: HandWaveBadge): ImageData {
+function drawEverythingToImage(nametag: NameTagBadge, handWave: HandWaveBadge, videoWidth: number, videoHeight: number): ImageData {
   const canvas = document.createElement('canvas');
   const context = canvas.getContext('2d')!;
-  canvas.width = 1600; // Width of the canvas
-  canvas.height = 900; // Height of the canvas
+  canvas.width = videoWidth; // Width of the canvas
+  canvas.height = videoHeight; // Height of the canvas
+
+  // Apply transformations to maintain proportions
+  const xFactor = videoWidth / 1600;
+  const yFactor = videoHeight / 900;
```

Comment:
I think there are places here where these xFactor and yFactor could be avoided by using relative sizes rather than specific pixel sizes.  See https://developer.mozilla.org/en-US/docs/Web/CSS/font-size